python-package/lightgbm/sklearn.py

```diff
@@ -519,18 +519,18 @@ def __init__(
         self.n_jobs = n_jobs
         self.importance_type = importance_type
         self._Booster: Optional[Booster] = None
-        self._evals_result = None
-        self._best_score = None
-        self._best_iteration = None
+        self._evals_result: _EvalResultDict = {}
+        self._best_score: _LGBM_BoosterBestScoreType = {}
```

Comment:
When a variable doesn't have an explicit type hint, `mypy` will infer the type based on the first assignment of that variable.

Since these will eventually be dictionaries, initializing them to dictionaries convinces `mypy` that code which doesn't check that they're `None` is ok.